tsdb/db_test.go

```diff
@@ -2872,3 +2872,90 @@ func TestChunkReader_ConcurrentReads(t *testing.T) {
 	}
 	testutil.Ok(t, r.Close())
 }
+
+// TestHeadFlushIntoBlock ensures that the head flush
+// creates a block that is ready for loading and
+// that flushing the storage does not cause data loss.
+// This test:
+// * opens a storage;
+// * appends values;
+// * flushes the storage wal/head; and
+// * queries the storage to ensure the samples are present from the flushed block.
+func TestHeadFlushIntoBlock(t *testing.T) {
```

Comment:
I thought it was for the Flush method. Shall we keep the name method specific to `TestCompactHead`?